src/ui/preview/ViewHistoricalDataAction.js

```diff
@@ -32,4 +32,12 @@ export default class ViewHistoricalDataAction extends PreviewAction {
         this.cssClass = 'icon-eye-open';
         this.hideInDefaultMenu = true;
     }
+
+    appliesTo(objectPath, viewContext = {}, options = {}) {
+        if (objectPath.length && options.viewHistoricalData) {
```

Comment:
I will change it so it uses the viewContext to make this decision. The reason we cant let the action decide wether its applicable in this case, is because this action is voluntarily used by some tables, and not others (in openmct and vista). So in order to make it reusable across different versions of Open MCT, we need to let the view make this decision, in this case.